pkg/filter/filter.go

```diff
@@ -46,23 +51,25 @@ func (aw *wrapWriter) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	return nil, nil, errors.New("the ResponseWriter doesn't support the Hijacker interface")
 }
 
-func NewAuthenticationFilter(ctx context.Context, managementContext *config.ScaledContext, auditWriter *audit.LogWriter, next http.Handler) (http.Handler, error) {
+func NewAuthenticationFilter(ctx context.Context, managementContext *config.ScaledContext, auditWriter *audit.LogWriter, next http.Handler, clusterManager *clustermanager.Manager) (http.Handler, error) {
 	if managementContext == nil {
 		return nil, fmt.Errorf("Failed to build NewAuthenticationFilter, nil ManagementContext")
 	}
 	auth := requests.NewAuthenticator(ctx, managementContext)
 
 	return &authHandler{
-		auth:        auth,
-		next:        next,
-		auditWriter: auditWriter,
+		auth:           auth,
+		next:           next,
+		auditWriter:    auditWriter,
+		clusterManager: clusterManager,
```

Comment:
Can you narrow down this dependency to some smaller interface.  In the past I believe the auth filter is fairly standalone (not a huge dependency on the rest of rancher). This makes it directly dependent on a huge part of Rancher.